experimental/utilities.py

```diff
@@ -0,0 +1,43 @@
+import cv2
+import numpy as np
+
+
+def get_corners_from_contour(contour: np.ndarray, corner_amount=4) -> None:
+    """Gets the corners of a contour.
+
+    cv2 contours found using the cv2.findContours() method are imperfect and not helpful for
+    use with tools like solvePNP. This method approximates the contour as a polygon with a specified
+    amount of corners/sides. It is more accurate than the previous version, as it uses the bisection method,
+    rather than just increasing or decreasing a number by a set amount.
+
+    NOTE: The points are in clockwise or anticlockwise depending on the input contour. 
+        The returned contour should be checked if the order matters.
+
+    Args:
+        contour: a single contour in the format used by opencv.
+        corner_amount: The amount of corners the approximated polygon should have.
+
+    Returns:
+        The approximated polygon, in a similar format to the 'contour' parameter.
+    """
+
+    arclength = cv2.arcLength(contour, True)
+    lower = 0 * arclength
+    depth = 15
+    upper = 2 * arclength
+
+    for _ in range(depth):
+        current = (lower + upper) / 2
+        hull = cv2.convexHull(cv2.approxPolyDP(contour, current, True))
+        if len(hull) > corner_amount:
+            lower = current
+        else:
+            upper = current
+    while len(hull) != corner_amount:
```

Comment:
I think you need to explain why this is broken into two parts. Why not just do it all in the while loop? They seem to do the same thing.